htlcswitch/link.go

```diff
@@ -376,10 +376,99 @@ type channelLink struct {
 	// UpdateAddHTLC.
 	isIncomingAddBlocked atomic.Bool
 
+	// flushHooks is a hookMap that is triggered when we reach a channel
+	// state with no live HTLCs.
+	flushHooks hookMap
```

Comment:
I don't mind doing that, but it does make me wonder why we have the link mutex at all. I figured it would be to handle these exact cases where we wanted to modify rows on this structure. However, you mentioned it isn't locked during changes to the underlying LightningChannel either.